### PR TITLE
[Merged by Bors] - feat: a functor from a small category to a filtered category factors through a small filtered category

### PR DESCRIPTION
**Pull request by TwoFX**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Filtered.lean [hunk 1]
```diff
@@ -487,6 +488,146 @@ theorem tulip {j₁ j₂ j₃ k₁ k₂ l : C} (f₁ : j₁ ⟶ k₁) (f₂ : j
...
+
+instance : Faithful (smallFilteredIntermediateInclusion F) :=
+  show Faithful ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Faithful.comp _ _
+
+noncomputable instance : Full (smallFilteredIntermediateInclusion F) :=
+  show Full ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Full.comp _ _
+
+/-- The factorization through a small filtered category is in fact a factorization, up to natural
+    isomorphism. -/
+noncomputable def smallFilteredIntermediateFactoringCompSmallFilteredIntermediateInclusion :
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Filtered.lean:618]: If you rename the above two defs to `factoring` and `inclusion` and put them in the namespace `SmallFilteredIntermediate`, then this def can get a much shorter name.

File: Mathlib/CategoryTheory/Filtered.lean [hunk 2]
```diff
@@ -803,6 +944,135 @@ theorem of_equivalence (h : C ≌ D) : IsCofiltered D :=
...
+
+instance : Faithful (smallCofilteredIntermediateInclusion F) :=
+  show Faithful ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Faithful.comp _ _
+
+noncomputable instance : Full (smallCofilteredIntermediateInclusion F) :=
+  show Full ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Full.comp _ _
+
+/-- The factorization through a small filtered category is in fact a factorization, up to natural
+    isomorphism. -/
+noncomputable def smallCofilteredIntermediateFactoringCompSmallCofilteredIntermediateInclusion :
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Filtered.lean:1063]: Idem.

**Review by jcommelin — COMMENTED** [review]
The file on filtered categories is getting quite long. Would you mind renaming it to `Filtered/Basic.lean` and adding your patch to a new file in the `Filtered/` directory?

File: Mathlib/CategoryTheory/Filtered/Basic.lean [hunk 3]
```diff
@@ -487,6 +488,146 @@ theorem tulip {j₁ j₂ j₃ k₁ k₂ l : C} (f₁ : j₁ ⟶ k₁) (f₂ : j
...
+
+instance : Faithful (smallFilteredIntermediateInclusion F) :=
+  show Faithful ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Faithful.comp _ _
+
+noncomputable instance : Full (smallFilteredIntermediateInclusion F) :=
+  show Full ((equivSmallModel _).inverse ⋙ fullSubcategoryInclusion _) from Full.comp _ _
+
+/-- The factorization through a small filtered category is in fact a factorization, up to natural
+    isomorphism. -/
+noncomputable def smallFilteredIntermediateFactoringCompSmallFilteredIntermediateInclusion :
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Filtered/Basic.lean:618]: Please use namespaces to make these names shorter.

File: Mathlib/CategoryTheory/Filtered/Basic.lean [hunk 4]
```diff
@@ -487,6 +488,146 @@ theorem tulip {j₁ j₂ j₃ k₁ k₂ l : C} (f₁ : j₁ ⟶ k₁) (f₂ : j
 
 end SpecialShapes
 
+section FilteredClosure
```
[the remark this answers is not in the document]
Comment (TwoFX) [reply Mathlib/CategoryTheory/Filtered/Basic.lean:491]: Hi, I have seen your previous comments, I just haven't found the time to apply them yet, which is why I removed `awaiting-review` and added `WIP` yesterday. I will change the labels back when I'm done addressing your comments.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**